SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -701,25 +701,26 @@ int32 USpatialNetDriver::ServerReplicateActors_ProcessPrioritizedActors(UNetConn
 				// or it's an editor placed actor and the client hasn't initialized the level it's in
 				if (Channel == NULL && GuidCache->SupportsObject(Actor->GetClass()) && GuidCache->SupportsObject(Actor->IsNetStartupActor() ? Actor : Actor->GetArchetype()))
 				{
-					// Create a new channel for this actor.
-					Channel = (USpatialActorChannel*)InConnection->CreateChannel(CHTYPE_Actor, 1);
-					if (Channel)
+					// If we're a singleton, and don't have a channel, defer to GSM
+					if (Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_Singleton))
 					{
-						if (Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_Singleton))
+						Channel = GlobalStateManager->AddSingleton(Actor);
+					}
+					else
+					{
+						// Create a new channel for this actor.
+						Channel = (USpatialActorChannel*)InConnection->CreateChannel(CHTYPE_Actor, 1);
+						if (Channel)
 						{
-							SingletonActorChannels.Add(Actor->GetClass(), TPair<AActor*, USpatialActorChannel*>(Actor, Channel));
+							Channel->SetChannelActor(Actor);
 						}
-						else
+						// if we couldn't replicate it for a reason that should be temporary, and this Actor is updated very infrequently, make sure we update it again soon
+						else if (Actor->NetUpdateFrequency < 1.0f)
```

Comment:
Context on this change?

Reply:
This is existing code, it's just coming up as changed because I altered the surrounding scope. The scope was altered because we now rely entirely on the GSM to create channels for singletons.